### PR TITLE
Make `LocalPolicies` immutable to avoid concurrent modify inconsistent.

### LINKED ISSUE #9595: Flaky-test: V1_AdminApiTest.testNamespaceSplitBundleConcurrent
V1_AdminApiTest is flaky. The `testNamespaceSplitBundleConcurrent` test method fails sporadically in **local environment**

[example failure](https://gist.github.com/WJL3333/de319f644e976d25f522c8b14f6cc1b9#file-gistfile1-txt)

i check some github Action log found that this less happen in CI env but fail in local env very often.

### PR DESCRIPTION
**Pull request by WJL3333**
Fixes #9595

### Motivation

make `LocalPolicies` immutable to avoid concurrent modify inconsistent.

`NamespaceService.updateNamespaceBundles` is not thread safe.
when split the same namespace concurrently,
both 2 thread can get the same `LocalPolicies` instance from pulsar.getLocalZkCacheService().policiesCache()

and `updateNamespaceBundles` change the LocalPolicies.bundles without
concurrent protect. which can cause the param `NamespaceBundles` is inconsistent with the serialized data.

eg. 

origin bundle data [1,3,5]

the thread1 split bundle [1,3] and want to set path in zk with bundle data [1, 2, 3,5] and version 3

the thread2 split bundle [3,5] and want to set path in zk with bundle data [1,3, 4, 5] and version 3

one concurrent run can be following: 
after thread1 gene bundle data [1,2,3,5] before go to the line zk.setData(version)
thread2 modify the bundle data [1,3,4,5] 
then thread1 won the version write. and return success `updateNamespaceBundles` but the data write is [1,3,4,5]

thread2 will retry and try to split [3,5] again.but [3,5] already not occur in bundleData (because modify the same instance concurrently).then the unit test in #9595 fail with bundle not found in namespace.

### Modifications

make `LocalPolicies` immutable. each time modify field need object copy to avoid change the same instance concurrently.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as *V1_AdminApiTest.testNamespaceSplitBundleConcurrent*.

### Does this pull request potentially affect one of the following parts:

no 

### Documentation

no

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java [hunk 1]
```diff
@@ -52,4 +60,12 @@ public boolean equals(Object obj) {
         return false;
     }
 
+    @Override
+    public String toString() {
```
Comment (merlimat) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:64]: Use `@ToString()` annotation on the class

Comment (merlimat) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:64]: Also we could similarly take out the `equals()` and `hashCode()` methods.

Comment (WJL3333) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:64]: thanks for reply. i'll change this.

**Review by zymap — APPROVED** [review]

**Comment by sijie** [comment]
Ping @merlimat

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java [hunk 2]
```diff
@@ -19,37 +19,34 @@
...
-import com.google.common.base.Objects;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
 
 /**
  * Local policies.
  */
+@ToString
+@EqualsAndHashCode
 public class LocalPolicies {
```
Comment (eolivelli) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:30]: can this change impact compatibility with existing clusters ?

Jackson Mapper uses Java reflection
How do we guarantee that the JSON code of this version is compatible with the one of the previous version ?

Can we add tests that try to deserialise a JSON produced with the old version ? or do we already have some ?

**Review by eolivelli — APPROVED** [review]
LGTM

I left one comment, can you please take a look ?

**Comment by zymap** [comment]
@WJL3333 Could you please take a look at Enrico's comment?

**Comment by zymap** [comment]
Because it's flaky tests fix. I will move this to the next release.

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java (hunk 2 above)
Comment (WJL3333) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:30]: yes, you are right. i'll add a test for this case.

Comment (WJL3333) [reply pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/LocalPolicies.java:30]: @eolivelli add a test case in `LocalPolicesTest.testMakeLocalPoliciesImmutableSerializationCompatibility`. can you take a look again ?

**Comment by codelipenghui** [comment]
@WJL3333 You can rebase to the current master branch to check if the CI can get passed.

**Comment by 315157973** [comment]
Some points are fixed in this pr https://github.com/apache/pulsar/pull/9900 . Please take a look @WJL3333

**Comment by WJL3333** [comment]
@codelipenghui @315157973  rebase current master. please check again

**Comment by eolivelli** [comment]
@merlimat can you take a final look please ?

**Comment by eolivelli** [comment]
This patch cannot be applied to branch-2.7, I am removing the release/2.7.2 label

**Pull request merged**